### PR TITLE
Feature/playlist track order

### PR DESCRIPTION
**Pull request by karrirasinmaki**
This PR tries to implement feature request #568.

Allows loading tracks of Spotify playlists in revered order, leading the newest track to be first in a list. Instead of paging tracks from start to end, with this feature, we can page tracks from end to start.

This PR contains:  
- Changes how Spotify APIs (playlist and tracks) is and can be called, and respective changes in actions and middleware.
- UI updates on the Playlist view to be able to choose between default and reverse sorting.

Hopefully, this PR serves as a good starting point to implement this feature in master branch.

### CODE REVIEW

File: src/js/services/spotify/actions.js [hunk 1]
```diff
@@ -1493,7 +1493,69 @@ export function savePlaylist(uri, name, description, is_public, is_collaborative
...
+          dispatch(coreActions.handleException(
+            'Could not load playlist',
+            error,
+          ));
+        },
+      );
+  };
+}
+
+export function getPlaylistFull(uri) {
```
Comment (karrirasinmaki) [inline src/js/services/spotify/actions.js:1558]: I kept the original ``getPlaylist`` function but changed its name.

File: src/js/util/format.js [hunk 2]
```diff
@@ -407,6 +407,7 @@ const formatPlaylist = function (data) {
     'last_modified',
     'can_edit',
     'owner',
+    'order',
```
Comment (karrirasinmaki) [inline src/js/util/format.js:410]: The ``order`` parameter is saved inside Playlist object.

**Comment by jaedb** [comment]
@karrirasinmaki nice work - you've put in some good thought into how this could be achieved.

The thing that I'm challenged by is that this would mean that the ordering of Spotify playlists differs from any other type of playlist. Wherever possible I've tried to keep logic as source-agnostic as possible.

Did you have any thoughts on how you might achieve this same logic for other providers?

**Comment by karrirasinmaki** [comment]
Thanks, @jaedb.

I think it's possible with other providers' APIs to fetch pages in reversed order too. I haven't taken a look at those yet - but am happy to!

Of course, it would be nice if all APIs would support filtering and ordering by different parameters. But at least Spotify API is limited in that sense. Client-side ordering would be a possibility but requires fetching the whole playlist in order to do so.

How is shuffle play implemented? Do you fetch the whole playlist in order to pick the next random song?

**Comment by jaedb** [comment]
Thanks @karrirasinmaki for your hard work on this! You'll be glad to know your feature is now in production. I took your approach into a bigger piece of work, so much thanks for your insight and contribution 👍

**Comment by karrirasinmaki** [comment]
That's great news @jaedb! Happy and honored to help :pray:

**Pull request closed without merging**